File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/CheckBox/Functionality/CheckStatePage.vb
Namespace CompanionKit.Controls.CheckBox.Functionality

	Public Class CheckStatePage

		Public Sub New()

			' This call is required by the Visual WebGui UserControl Designer.
			InitializeComponent()

			'Update label text with initial state of demonstrated CheckBox
			UpdateLabelText()

		End Sub

        ''' <summary>
        ''' Handles the Click event of the mobjCheckStateButton control.
        ''' </summary>
        ''' <param name="sender">The source of the event.</param>
        ''' <param name="e">The <see cref="System.EventArgs"/> instance containing the event data.</param>
        Private Sub mobjCheckStateButton_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles mobjCheckStateButton.Click
            UpdateLabelText()
        End Sub


		''' <summary>
		''' Updates label text, that indicates state of demonstrated CheckBox.
		''' </summary>
		''' <remarks></remarks>
		Private Sub UpdateLabelText()
            Me.mobjStateLabel.Text = String.Format("CheckBox state is {0}!", Me.mobjCheckBox.CheckState)
		End Sub
	End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Controls/CheckBox/Functionality/CheckStatePage.cs
#region Using

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms;

#endregion

namespace CompanionKit.Controls.CheckBox.Functionality
{
    public partial class CheckStatePage : UserControl
    {
        
        public CheckStatePage()
        {
            InitializeComponent();

            //Update label text with initial state of demonstrated CheckBox
            UpdateLabelText();
        }

        /// <summary>
        /// Handles the Click event of the mobjCheckStateButton control.
        /// </summary>
        /// <param name="sender">The source of the event.</param>
        /// <param name="e">The <see cref="EventArgs"/> instance containing the event data.</param>
        private void mobjCheckStateButton_Click(object sender, EventArgs e)
        {
            UpdateLabelText();
        }

        /// <summary>
        /// Updates label text, that indicates state of demonstrated CheckBox.
        /// </summary>
        private void UpdateLabelText()
        {
            this.mobjStateLabel.Text = String.Format("CheckBox state is {0}!", mobjCheckBox.CheckState);
        }
    }
}